modules/io/src/common/xviz-data.js

```diff
@@ -35,15 +35,16 @@ import {XVIZ_FORMAT} from './constants';
 // - arraybuffer which is a JSON string
 // - JSON object
 // - arraybuffer which is a GLB
+// opts.messageType is the XVIZ Envelope 'type', i.e. one of ('xviz/state_update', 'xviz/metadata', etc.)
 export class XVIZData {
-  constructor(data) {
+  constructor(data, messageType) {
```

Comment:
I'm wondering if we should put this in an object?

`constructor(data, options) {`

Reply:
updated